instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/patch.py

```diff
@@ -0,0 +1,136 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# pylint:disable=relative-beyond-top-level
+import redis
+from wrapt import ObjectProxy, wrap_function_wrapper
+
+from opentelemetry import trace
+
+from .util import _extract_conn_attributes, _format_command_args
+from .version import __version__
+
+_DEFAULT_SERVICE = "redis"
+_RAWCMD = "db.statement"
+_CMD = "redis.command"
+
+
+def patch(tracer=None):
```

Comment:
Maybe it's a good idea to update right now to take a TracerProvider as described in https://github.com/open-telemetry/opentelemetry-python/issues/585. Up to you.